src/js/base/module/Editor.js

```diff
@@ -357,6 +357,11 @@ export default class Editor {
         }
       }
       this.setLastRange();
+
+      // record undo in the key event except keyMap.
+      if (this.hasKeyShortCut === false) {
```

Comment:
This should be run by an optional parameter. (e.g. `historyLevel` or `recordEveryKeystroke`)

Reply:
And it would be good to have a length option for the history stack.

Reply:
> And it would be good to have a length option for the history stack.

Resolved by https://github.com/summernote/summernote/pull/3563